provisioning/virtual-machines/ansible/roles/gns3/tasks/main.yml

```diff
@@ -75,3 +75,114 @@
     owner: "{{ ansible_user }}"
     group: "{{ ansible_user }}"
     mode: '0644'
+
+### The following Ansible tasks replace the previous manual setup of the GNS3 environment
+
+## 1.  Download the Cisco IOS Image
+
+# 1.1 Download nfs-common
+- name: Install nfs utils
+  apt:
+    name: nfs-common
+    state: present
+
+# 1.2 Mount GNS4 NFS install volume
+- name: Mount GNS3 NFS install volume
+  ansible.posix.mount:
+    src: "{{ gns3_device_images }}"
+    path: "/mnt/gns3_images"
+    opts: rw,sync,hard
+    boot: false
+    fstype: nfs
+    state: mounted
+
+# 1.3 Decompress the file from .bin to .image
+- name: Unzip the file
+  shell: >
+        unzip -p /mnt/gns3_images/c2691-advipservicesk9-mz.124-25c.bin >
+        /home/{{ ansible_user }}/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.image
+  ignore_errors: yes
+
+ # 1.4 Unmount GNS3 NFS install volume
+- name: Unmount Matlab NFS install volume
+  ansible.posix.mount:
+    src: "{{ gns3_device_images }}"
+    path: "/mnt/gns3_images"
+    opts: rw,sync,hard
+    boot: false
+    fstype: nfs
+    state: absent
+
+## 2.  Create a new template for the Cisco 2691 router
+
+# GNS3 server is usually launched through UI. Here we need to start it via command line.
+
+# 2.1 Create a gns3.service file which is going to be handled via systemd
+- name: Copy the gns3 service file
+  copy:
+    src: "{{ role_path }}/files/gns3.service"
+    dest: /etc/systemd/system/gns3.service
+    mode: '0755'
+
+# 2.2 Enable and start the new gns3 service
+- name: Enable and start the gns3 service
+  systemd:
+    name: gns3
+    state: started
+    enabled: yes
+
+# 2.3 The GNS3 server needs some seconds to be ready
+- name: Introduce a sleep delay of 5 seconds
+  pause:
+    seconds: 5
+
+# 2.4 Create the request body to be used in Cisco C2961 template creation
+- name: Copy Cisco 2961 template request body
+  copy:
+    src: "{{ role_path }}/files/cisco-c2961.json"
+    dest: /tmp/cisco-c2961.json
+    mode: '0644'
+
+# 2.5 Performing the request to the GNS3 server
+- name: Create new template using GNS3 API
+  command: sudo  curl -X POST "http://localhost:3080/v2/templates" -d @/tmp/cisco-c2961.json
+
+# 2.5 Remove the temporary file used to hold the request body
+- name: Remove JSON request file 
+  file:
+    path: /tmp/json_file.json
+    state: absent
```

Comment:
You might probably manage to do this with a single shot with Ansible uri module
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/uri_module.html